solver/llbop/exec.go

```diff
@@ -82,6 +82,8 @@ func (e *execOp) Run(ctx context.Context, inputs []solver.Ref) ([]solver.Ref, er
 				return nil, errors.Errorf("invalid reference for exec %T", inputs[int(m.Input)])
 			}
 			mountable = ref
+		} else {
+			return nil, errors.Errorf("mount %s has no input", m.Dest)
```

Comment:
I think the case you are looking for is when there is no input and output. Having no input with an output is used for "scratch" layers. Or just check `mountable==nil` after these conditions. Fine for me to have a second check in ocispec generator as well.